store/store.go

```diff
@@ -765,3 +765,120 @@ func (s *SnapUbuntuStoreRepository) SuggestedCurrency() string {
 	}
 	return s.suggestedCurrency
 }
+
+// Optionally specify parameters to the store purchase.
+// Currency: ISO 4217 code as string
+// BackendID: The payment backend to use, e.g. "credit_card", "paypal".
+// MethodID: The unique identifier for the payment method, retrieved from the pay server.
+type BuyOptions struct {
+	Currency  string
+	BackendID string
+	MethodID  int64
+}
+
+// When the purchase state is "InProgress" (i.e. requires user interaction to complete)
+// this struct contains the information required to complete the purchase.
+type BuyRedirect struct {
+	RedirectTo string
+	PartnerID  string
+}
+
+// purchaseInstruction encapsulates the data that must be sent in order to make a purchase from the store.
+// The Device ID can be sent in the extended header "X-Device-Id".
+// The Partner ID (e.g. "bq") can be sent in the extended header "X-Partner-Id".
+type purchaseInstruction struct {
+	SnapID    string  `json:"snap_id"`
+	ItemSKU   string  `json:"item_sku,omitempty"`
+	Amount    float64 `json:"amount,omitempty"`
+	Currency  string  `json:"currency,omitempty"`
+	BackendID string  `json:"backend_id,omitempty"`
+	MethodID  int64   `json:"method_id,omitempty"`
+}
+
+type buyError struct {
+	ErrorMessage string `json:"error_message"`
+}
+
+// Buy the specified snap using the default currency and payment method.
+// Returns the state of the purchase: Complete, Cancelled, InProgress or Pending.
```

Comment:
Needs tweaking after the BuyResult suggestion above. Typical Go doc wording would also would need a verb after the method name, which sounds awkward in that specific case:

`// Buy buys ...`

So we can tweak a bit perhaps:

`// Buy sends a purchase request ...`